### PR TITLE
Add Foreman and Overview docs

### PR DESCRIPTION
**Pull request by ehelms**
Replaces:

 * https://projects.theforeman.org/projects/foreman/wiki/Infrastructure

This is scant information to start with as I am porting the wiki pages over first for review, and to allow brain dumps of knowledge.

### CODE REVIEW

File: docs/foreman-puppetserver.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Puppetmaster and Foreman
+
+Puppet and Foreman are of course used to manage the machines. The Foreman instance is accessible only to those with SSH access to puppet.theforeman.org. Add the following snippet to ~/.ssh/config:
```
Comment (ekohl) [inline docs/foreman-puppetserver.md:3]:
```suggestion
Puppet and Foreman are of course used to manage the machines. The Foreman instance is accessible only to those with SSH access to puppetmaster.theforeman.org. Add the following snippet to `~/.ssh/config`:
```

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Overview
+
+The Foreman project runs a number of different servers for testing, packaging, and continuous integration. Most of these servers are available as Jenkins build slaves for http://ci.theforeman.org.
```
Comment (ekohl) [inline docs/overview.md:3]:
```suggestion
The Foreman project runs a number of different servers for testing, packaging, and continuous integration. Most of these servers are available as Jenkins build nodes for https://ci.theforeman.org.
```

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+| ci.theforeman.org | OSUOSL | Jenkins |
+|   |   |   |
+|   |   |   |
+
+## Infrastructure providers
+
+A list of the hosting we have, who provides it, and what capacity it has
+
+* Rackspace
+** $2k per month
```
Comment (ekohl) [inline docs/overview.md:23]: No longer sponsored so $0?

File: docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+** Hosts web02, the Elastic logging instance, Jenkins, and some permanent slaves
```
Comment (ekohl) [inline docs/overview.md:24]: I think the logging instance and slaves were removed, but the Foreman instance is there as well

File: docs/overview.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+** Contact support from our account as needed
+* Scaleway
+** $25 per month (not a lot!), but they're cheap so we do OK. Hosts Discourse, Redmine, and some ARM build slaves
```
Comment (ekohl) [inline docs/overview.md:27]: I think this sponsoring has gone away so $0?

File: docs/overview.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+** Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts
+* Fastly
+** $1k/month CDN
```
Comment (ekohl) [inline docs/overview.md:30]: We should mention that this is sponsored

File: docs/overview.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+** Elaine Greenberg was initial contact
+* OSUOSL
+** Free
```
Comment (ekohl) [inline docs/overview.md:33]:
```suggestion
** Sponsored
```

File: docs/overview.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+** Hosts test machines, slaves, and stats.tf.o
```
Comment (ekohl) [inline docs/overview.md:34]:
```suggestion
** Hosts test machines, Jenkins nodes and web01
```

File: docs/overview.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+** Contact Lance Alberston if more capacity is needed
+* NETWAYS
+** Free
```
Comment (ekohl) [inline docs/overview.md:37]: Sponsored

File: docs/overview.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+** Contributing a single VM for the Builders/RH hostgroup
```
Comment (ekohl) [inline docs/overview.md:38]: We actually have an account in their openstack instance with capacity for more than one. I just had issues installing and haven't gotten back to their support.

File: docs/overview.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+** Option to add an Icinga Monitoring host here, talk to Dirk to progress
```
Comment (ekohl) [inline docs/overview.md:39]: Currently blocked on me responding to a support ticket

**Review by ekohl — CHANGES_REQUESTED** [review]
You missed Gandi who sponsors our theforeman.org domain.

File: docs/overview.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs. Fork http://github.com/theforeman/foreman-infra and add an ssh_user resource to the users module (see puppet/modules/users/manifests/init.pp) and your key into the files directory. Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and update the puppetmaster.
+
+## Landscape
+
+
+| Host | Provider | Function |
+|---|---|---|
+| ci.theforeman.org | OSUOSL | Jenkins |
+|   |   |   |
+|   |   |   |
```
[the remark this answers is not in the document]
Comment (ekohl) [reply docs/overview.md:16]: I think listing the individual hosts is not that important and what we have Foreman for. IMHO a list of roles (foreman, web, Jenkins master and Jenkins node) is more important. They can then link to the individual detailed pages.

File: docs/overview.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+## Access
+
+Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs. Fork http://github.com/theforeman/foreman-infra and add an ssh_user resource to the users module (see puppet/modules/users/manifests/init.pp) and your key into the files directory. Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and update the puppetmaster.
+
+## Landscape
+
+| Role | Provider |
+|---|---|
+| [Jenkins](./docs/jenkins.md) | OSUOSL |
+| [Koji](./docs/koji.md) | AWS |
```
Comment (ekohl) [inline docs/overview.md:14]: While these have a single provider, Jenkins nodes won't.

File: docs/overview.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
+# Overview
+
+The Foreman project runs a number of different servers for testing, packaging, and continuous integration.
+
+## Access
+
+Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs. Fork http://github.com/theforeman/foreman-infra and add an ssh_user resource to the users module (see puppet/modules/users/manifests/init.pp) and your key into the files directory. Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and update the puppetmaster.
```
Comment (evgeni) [inline docs/overview.md:7]:
```suggestion
Access to Foreman project infrastructure is available for those who wish to assist in building packages, testing, and building Jenkins jobs.

Fork https://github.com/theforeman/foreman-infra and add your key into the [files directory](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/users/files) of the [`users` module](https://github.com/theforeman/foreman-infra/blob/master/puppet/modules/users/). Submit a pull request to the infrastructure project and then talk to ? in #theforeman-dev on irc.freenode.net. One of them can merge your change and add your user in the Foreman web UI.
```

File: docs/overview.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+|   |   |
+
+## Infrastructure providers
+
+A list of the hosting we have, who provides it, and what capacity it has
+
+* Rackspace
+** 0k/month
+** Hosts web02, Jenkins
+** Contact support from our account as needed
```
Comment (evgeni) [inline docs/overview.md:24]:
```suggestion
* Rackspace
  * 0k/month
  * Hosts web02, Jenkins, Foreman/Puppet
  * Contact support from our account as needed
```
the formating applies to the whole section  ;-)

File: docs/overview.md [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+* Scaleway
+** 0k/month
+** Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts
```
Comment (evgeni) [inline docs/overview.md:27]:
```suggestion
  * Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts
  * Hosts Discourse, Redmine, ARM Jenkins nodes (EOL)
```

File: docs/overview.md [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+* OSUOSL
+** Sponsored
+** Hosts test machines, Jenkins nodes, and web01
+** Contact Lance Alberston if more capacity is needed
+* NETWAYS
+** Sponsored
+** Openstack instance for spinning up as needed compute
+** Option to add an Icinga Monitoring host here, talk to Dirk to progress
+* Gandi
+** Provides DNS
```
Comment (evgeni) [inline docs/overview.md:41]: sponsors theforeman.org domain

File: docs/overview.md (hunk 14 above)
Comment (ekohl) [reply docs/overview.md:7]: Perhaps a Discourse post is a better suggestion?

Comment (evgeni) [reply docs/overview.md:7]: Also that, yeah.

File: docs/overview.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+A list of the hosting we have, who provides it, and what capacity it has
+
+* Rackspace
+  * 0k/month
+  * Hosts web02, Jenkins
+  * Contact support from our account as needed
+* Scaleway
+  * 0k/month
+  * Support usually helpful, Edouard Bonlieu & Yann Léger were initial contacts
+    * Hosts Discourse, Redmine, ARM Jenkins nodes
```
Comment (evgeni) [inline docs/overview.md:32]:
```suggestion
  * Hosts Discourse, Redmine, ARM Jenkins nodes
```

**Comment by ehelms** [comment]
Is there anything missing to merge this?

File: docs/overview.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+| [Jenkins](./docs/jenkins.md) | OSUOSL |
+| [Koji](./docs/koji.md) | AWS |
+| [Jenkins Nodes](./docs/jenkins.md)  | OSUOSL / AWS |
+
+## Infrastructure providers
+
+A list of the hosting we have, who provides it, and what capacity it has
+
+* Rackspace
+  * 0k/month
```
Comment (ekohl) [inline docs/overview.md:26]: This is a bit odd
```suggestion
  * Previously sponsored, [being phased out](https://community.theforeman.org/t/rfc-moving-off-of-rackspace-infrastructure/17932)
```

File: docs/overview.md [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+  * Hosts web02, Jenkins
+  * Contact support from our account as needed
+* Scaleway
+  * 0k/month
```
Comment (ekohl) [inline docs/overview.md:30]:
```suggestion
  * Previously sponsored, planned to be phased out
```

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by ehelms** [comment]
This has addressed all current comments. I would really like to get it merged and iterate from there.

File: docs/overview.md [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+| [Koji](./docs/koji.md) | AWS |
+| [Jenkins Nodes](./docs/jenkins.md)  | OSUOSL / AWS |
+
+## Infrastructure providers
+
+A list of the hosting we have, who provides it, and what capacity it has
+
+* Rackspace
+  * Previously sponsored, [being phased out](https://community.theforeman.org/t/rfc-moving-off-of-rackspace-infrastructure/17932)
+  * Hosts web02, Jenkins
```
Comment (ekohl) [inline docs/overview.md:25]: web02 was turned off

**Review by ekohl — APPROVED** [review]

**Pull request merged**